astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -536,6 +536,67 @@ def divide(self, operand, operand2=None, **kwargs):
         return self._prepare_then_do_arithmetic(np.true_divide, operand,
                                                 operand2, **kwargs)
 
+    def pow(self, exponent, propagate_uncertainties=True, **kwargs):
+        """Raises ``self`` to the power of an exponent.
+
+        Non-integer and non-positive exponents are not supported
+        if uncertainties must be propagated.
+        **kwargs are passed to NDData constructor.
+
+        Parameters
+        ----------
+        exponent: `numbers.Integral`
+            The exponent to which ``self`` is raised.  Must be > 0
+
+        propagate_uncertainties : `bool` or ``None``, optional
+            If ``None`` the result will have no uncertainty. If ``False`` the
+            result will have a copied version of the first operand that has an
+            uncertainty. If ``True`` the result will have a correctly propagated
+            uncertainty from the uncertainties of the operands but this assumes
+            that the uncertainties are `NDUncertainty`-like. Default is ``True``.
+
+        Returns
+        -------
+        result : `~astropy.nddata.NDData`-like
+            The resulting dataset
+        """
+
+        if (self.uncertainty is not None and propagate_uncertainties is True and
+                not (isinstance(exponent, numbers.Integral) and exponent > 0)):
+            raise TypeError("Non-positive and non-integer exponents not supported "
+                            "if propagate_uncertainties is True.")
+        # If exponent is 1, NDData is unchanged.
+        if exponent == 1 and isinstance(propagate_uncertainties, bool):
+            if kwargs.get("copy", False) is True:
```

Comment:
I think this could just be:

```suggestion
            if kwargs.get("copy", False):
```